src/renovate.ts

```diff
@@ -0,0 +1,64 @@
+import Docker from './docker';
+import { exec } from '@actions/exec';
+import fs from 'fs';
+import path from 'path';
+
+class Renovate {
+  private configFileEnv = 'RENOVATE_CONFIG_FILE';
+  private tokenEnv = 'RENOVATE_TOKEN';
+  private configFileMountDir = '/github-action';
+
+  private configFile: string;
+  private docker: Docker;
+
+  constructor(configFile: string, private token: string) {
+    this.configFile = path.resolve(configFile);
+
+    this.validateArguments();
+
+    this.docker = new Docker();
+  }
+
+  runDockerContainer(): Promise<void> {
```

Comment:
I would prefer an async function  with simple try catch here.

Reply:
What is the reasoning behind this? What's wrong with promises?